src/webview/common-ext/utils.ts

```diff
@@ -82,6 +86,67 @@ export function validateURL(event: Message | { command: string; data: object },
     } as validateURLProps
 }
 
+function searchBuilderImages(directory: Uri) {
+    const filesToCheck = [
+        'Dockerfile',
+        '.github/workflows/**/*.yml',
+        '.gitlab-ci.yml',
+        'Jenkinsfile',
+        'docker-compose.yml',
+        'build.sh',
+        'build.gradle',
+        'pom.xml'
+    ];
+
+    const builderImages:{file: string, matches: RegExpMatchArray}[] = [];
+
+    filesToCheck.forEach((pattern) => {
+        const files = glob.sync(path.join(directory.fsPath, pattern).replace(/\\/g, '/'));
+
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
```

Comment:
This is a legitimate problem picked up by eslint.

The `builderImages` array might be returned before all the files are analyzed.

To prevent this, you need to await the result of the async function. eg.
```typescript
await Promises.all(files.map(async (file) => {
  // ... existing code ...
}
```

This means that you also need to make the enclosing two functions async as well, eg.

```typescript
filesToCheck.forEach((pattern) =>
// ...
```
becomes

```typescript
await Promises.all(filesToCheck.map((pattern) => {
// ...
```
and
```typescript
function searchBuilderImages(directory: Uri) {
//...
```

becomes

```typescript
async function searchBuilderImages(directory: Uri) {
//...
```